### PR TITLE
All programs page & rename/archive/delete functionality

### PR DESCRIPTION
**Pull request by brittanylau**
This PR implements the all programs page, and admin functionality for organization admins:
- create program (frontend only)
- rename program
- archive/unarchive program
- delete program

Screen Shot 2020-12-15 at 2 34 42 PM

This also includes some updates to our models:
- removed the programs array from organizations, and changed it to be a reference on the programs model
- added `archived` and `deleted` fields to the programs model
- changed `adminOrganizations` on the users model to be a single ObjectId instead of an array (a user can only be an admin of one org based on the designs)

### Up next

I've created tickets for what's left for this page, but won't tackle them in this PR because it's getting pretty big.

### CODE REVIEW

File: backend/models/users.js [hunk 1]
```diff
@@ -35,8 +35,8 @@ const userSchema = new mongoose.Schema(
     programs: {
       type: [programRole]
     },
-    adminOrganizations: {
-      type: [mongoose.SchemaTypes.ObjectId]
+    adminOrganization: {
+      type: mongoose.SchemaTypes.ObjectId
```
Comment (brittanylau) [inline backend/models/users.js:39]: Based on the UI, a user should only be an admin of max one organization

File: backend/routes/organizations.js [hunk 2]
```diff
@@ -1,27 +1,13 @@
...
-  db["Authentication"].organizations
-    .find()
-    .then(function(found) {
-      res.json(found);
-    })
-    .catch(function(err) {
-      res.send(err);
-    });
-});
-
```
Comment (brittanylau) [inline backend/routes/organizations.js:24]: This function is unused!

File: backend/routes/programs.js [hunk 3]
```diff
@@ -25,28 +23,73 @@ router.get("/all", function(req, res) {
...
-    })
-    .catch(function(err) {
-      res.send(err);
-    });
+  try {
+    // TODO: call programsServices.createProgram()
+    res.status(201).send();
+  } catch (e) {
+    res.send(e);
+  }
```
Comment (brittanylau) [inline backend/routes/programs.js:33]: Just a placeholder for now, this should be implemented in a future PR when the ticket becomes unblocked

File: backend/routes/users.js [hunk 4]
```diff
@@ -117,17 +150,4 @@ router.put("/set-program-memberships", function(req, res) {
     });
 });
 
-//Update a user (Not sued for creating a new user)
-router.post("/", function(req, res) {
```
Comment (brittanylau) [inline backend/routes/users.js:121]: Also unused

**Comment by brittanylau** [comment]
This PR could also use a bit of CSS help!
* the program menu shows up in the top left corner (instead of next to the button)
* underlines on the "view" button :(
* the "view" buttons should be lined up when I'm part of programs that are also outside my organization
Screen Shot 2020-12-15 at 2 41 22 PM

**Comment by ahmedhamodi** [comment]
@brittanylau - see this PR on how to remove the underline on the View button text:
https://github.com/uwblueprint/cookie-monster/pull/286

Believe it's just text-decoration: none!

**Review by GodGreg — APPROVED** [review]

**Pull request merged**